packages/amplify_datastore_plugin_interface/lib/src/types/utils/parsers.dart

```diff
@@ -16,11 +16,11 @@
 // ignore_for_file: public_member_api_docs
 
 // only to be used internally by amplify-flutter library
-import 'package:collection/collection.dart' show IterableExtension;
-
+import 'package:collection/collection.dart' show IterableExtension;
+
 String? enumToString(Object? o) =>
     o != null ? o.toString().split('.').last : null;
 
 // only to be used internally by amplify-flutter library
-T? enumFromString<T>(String key, List<T> values) =>
-    values.firstWhereOrNull((v) => key == enumToString(v));
+T? enumFromString<T>(String? key, List<T> values) =>
+    key == null ? null : values.firstWhereOrNull((v) => key == enumToString(v));

```

Comment:
Can't verify from my phone but it seems like the ternary isn't even needed unless there are compiler complaints about trying to compare a nullable String. But this is nitpicky anyway

Reply:
Yeah, you're right. It was to protect against the case where the enum has a case named `null`, but I just realized Dart doesn't let you do that anyway.